hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -187,6 +187,18 @@ public final class HoodieMetadataConfig extends HoodieConfig {
       .sinceVersion("0.11.0")
       .withDocumentation("Comma-separated list of columns for which column stats index will be built. If not set, all columns will be indexed");
 
+  public static final String COLUMN_STATS_INDEX_PROCESSING_MODE_IN_MEMORY = "in-memory";
+  public static final String COLUMN_STATS_INDEX_PROCESSING_MODE_SPARK = "spark";
```

Comment:
nit: this can be enums.

Reply:
`COLUMN_STATS_INDEX_PROCESSING_MODE_SPARK`: based on the logic, this mode is more of leveraging engine context, could be Spark or Flink (plain Java), right?

Reply:
Good call. This shouldn't be Spark this should an engine-neutral